paginator/paginator.go

```diff
@@ -134,6 +134,10 @@ func (p *Paginator) setup(db *gorm.DB, dest interface{}) {
 		if rule.NULLReplacement != nil {
 			rule.SQLRepr = fmt.Sprintf("COALESCE(%s, '%v')", rule.SQLRepr, rule.NULLReplacement)
 		}
+		// cast custom types to their real underlying SQL type
+		if rule.CustomType != nil {
+			rule.SQLRepr = fmt.Sprintf("(%s)::%s", rule.SQLRepr, rule.CustomType.SQLType)
```

Comment:
Would it be better to use `CAST` function to cast the type? As my understanding `CAST` is widely shared in most sql databases. But if `::` is common too, then it is fine to keep it 👍

Reply:
It seems that `CAST` is more general and `::` seems to be PostgreSQL specific. Let's go with CAST.